model/models/mistral3/model_vision.go

```diff
@@ -0,0 +1,186 @@
+package mistral3
+
+import (
+	"math"
+
+	"github.com/ollama/ollama/fs"
+	"github.com/ollama/ollama/ml"
+	"github.com/ollama/ollama/ml/nn"
+)
+
+var batchSize int = 1
+
+func rotateHalf(ctx ml.Context, t ml.Tensor) ml.Tensor {
```

Comment:
I have a hack that allow reusing `ggml_rope` to do 2D rope, if you want to take a look: https://github.com/ngxson/llama.cpp/blob/dc0b448c8f5d16528ed55a88d761f907d4d25f66/examples/llava/clip.cpp#L561

Despite being a hack, my implementation should use less memory while being more performance than doing this way